packages/terra-list/src/ListItem.jsx

```diff
@@ -68,7 +68,7 @@ const ListItem = ({
   }
 
   return (
-    <li {...customProps} className={listItemClassNames} ref={refCallback}>
+    <li {...customProps} role="option" aria-selected={isSelected} className={listItemClassNames} ref={refCallback}>
```

Comment:
Not every list item will be a role of `option`. For the base list item, I'd keep the role as its default HTML semantics, e.g. no role, and then implement the role="option" in the selectable list items: 
* https://github.com/cerner/terra-core/blob/8c34cdca8c56eaaeb16b26a2b780e97934f662aa/packages/terra-list/src/MultiSelectList.jsx#L71
* https://github.com/cerner/terra-core/blob/8c34cdca8c56eaaeb16b26a2b780e97934f662aa/packages/terra-list/src/SelectableList.jsx#L70
* https://github.com/cerner/terra-core/blob/8c34cdca8c56eaaeb16b26a2b780e97934f662aa/packages/terra-list/src/SingleSelectList.jsx#L69

We'll want logic that defaults to "option" in the selectable list but takes a custom role for when we use the selectable lists in terra-menu.